### ISSUE TITLE
`@typescript-eslint/no-unused-vars` Complains when import is only used in jsdoc

### ISSUE THREAD

**Issue by ej612** [typescript-eslint/typescript-eslint#9435]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Repro Code

Hi there!

I have the following code:
```ts
/**
 * Bla bla {@link MyClass}
 */
export interface MyInterface {
}
```
I want my jsdoc to include a reference to the class in question, which means I have to import it:
```ts
import type { MyClass } from './MyClass';

/**
 * Bla bla {@link MyClass}
 */
export interface MyInterface {
}
```
This works, meaning the link is now properly resolved. Also, tsc understands that I'm using the import and doesn't complain. If I remove the link but keep the import, I get the following error (as expected):
```
'MyClass' is declared but its value is never read. ts(6133)
```
So everything's fine on the tsc front. However, `no-unused-vars` doesn't realize I'm using the import in a jsdoc comment, and complains:
```
'MyClass' is defined but never used. eslint(@typescript-eslint/no-unused-vars)
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    'no-unused-vars': 'off',
    '@typescript-eslint/no-unused-vars': ['error', {
        args: 'after-used',
        argsIgnorePattern: '^_',
        destructuredArrayIgnorePattern: '^_'
    }],
  },
};
```

### Expected Result

I would expect `no-unused-vars` to realize that I'm using the import - albeit only in a jsdoc comment - like tsc does.

### Actual Result

`no-unused-vars` sees that I'm not using the import in the code and complains.

**Comment by bradzacher** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/8902, #5017, #6967, #7944, #8258

Please use the search.
